tx-pool/src/component/pending.rs

```diff
@@ -56,11 +56,16 @@ impl PendingQueue {
     }
 
     // fill proposal txs
-    pub fn fill_proposals(&self, limit: usize, proposals: &mut HashSet<ProposalShortId>) {
+    pub fn fill_proposals(
+        &self,
+        limit: usize,
+        filter: &HashSet<ProposalShortId>,
+        proposals: &mut HashSet<ProposalShortId>,
+    ) {
         for id in self.sorted_keys() {
             if proposals.len() == limit {
                 break;
-            } else if proposals.contains(&id) {
+            } else if filter.contains(&id) || proposals.contains(&id) {
                 // implies that ancestors are already in proposals
                 continue;
             }
```

Comment:
If the proposal is in uncle already, we should continue to handle the ancestor of it, not just skip simply?

Reply:
You are right. We should continue to handle its ancestors.
I will fix it and try to add a unit test soon.

Reply:
Adding a unit test to cover `handle its ancestors` is out of this PR.